test/cpp/tensorexpr/test_llvm.cpp

```diff
@@ -1610,6 +1610,133 @@ TEST(LLVM, RFactorVectorizedReduction) {
   ExpectAllNear(b_v, b_ref, 1e-5);
 }
 
+TEST(LLVM, SimpleParallel) {
+  for (int test_cfg = 0; test_cfg < 4; test_cfg++) {
+    // Compute a simple operation, and try all loop-axis combination to be
+    // parallel or sequential.
+    ExecutionCounter counter(llvm_codegen_parallel_dispatched);
+    KernelScope kernel_scope;
+    const int M = 4;
+    const int N = 6;
+    Tensor* tensor = Compute(
+        "f", {{M, "m"}, {N, "n"}}, [](const VarHandle& m, const VarHandle& n) {
+          return cast<float>(m + n);
+        });
+    LoopNest loop_nest({tensor});
+    auto const& loops = loop_nest.getLoopStmtsFor(tensor);
+    For* m = loops[0];
+    For* n = loops[1];
+    if (test_cfg & 0x1) {
+      m->set_parallel();
+    }
+    if (test_cfg & 0x2) {
+      n->set_parallel();
+    }
+    loop_nest.prepareForCodegen();
+    Stmt* stmt = loop_nest.root_stmt();
+    Placeholder f_buf(BufHandle(tensor->buf()));
+    LLVMCodeGen cg(stmt, {f_buf});
```

Comment:
Placeholder is not needed here, you could pass `tensor` directly to the codegen constructor:
```
LLVMCodeGen cg(stmt, {tensor});
```